src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs

```diff
@@ -647,6 +647,26 @@ class N { }
             edits.VerifyRudeDiagnostics();
         }
 
+        [Fact, WorkItem(48628, "https://github.com/dotnet/roslyn/issues/48628")]
+        public void ModifiersUpdate_IgnoreUnsafe2()
+        {
+            var srcA1 = "partial class C { unsafe void F() { } }";
+            var srcB1 = "partial class C { }";
+            var srcA2 = "partial class C { }";
+            var srcB2 = "partial class C { void F() { } }";
+
+            EditAndContinueValidation.VerifySemantics(
+                new[] { GetTopEdits(srcA1, srcA2), GetTopEdits(srcB1, srcB2) },
+                new[]
+                {
+                    DocumentResults(),
+                    DocumentResults(semanticEdits: new[]
+                    {
+                        SemanticEdit(SemanticEditKind.Update, c => c.GetMember<INamedTypeSymbol>("C").GetMember("F"))
+                    }),
+                });
```

Comment:
Yep, this looks good to me. One of the parameters to `DocumentResults` is a list of diagnostics to verify, and since you've passed none it will verify that none exist.

Reply:
Looks good, but we'll need a test for properties as well.

This will also need to be updated:
https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs#L2642-L2650

Reply:
@Youssef1313 I'd be fine with merging your PR as is, without handling the properties case. I'd then follow up on that as a separate issue. Sounds good?

Reply:
@tmat Yeah that would be good. Thanks. I was a bit stuck with how the properties case should be handled due to lack of knowledge about EnC analyzer.

Reply:
OK. Yeah, I realized it would be quite tricky to implement, so better not block your PR for that corner case.

Reply:
So let's file a new issue for the properties, add a test and link it to this issue and update https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs#L2642-L2650 to link to that issue as well. Then we can merge.

Reply:
Opened https://github.com/dotnet/roslyn/issues/52300